sdk/common/src/main/java/io/opentelemetry/sdk/common/export/CompletableResultCode.java

```diff
@@ -107,4 +109,42 @@ public CompletableResultCode whenComplete(Runnable action) {
     }
     return this;
   }
+
+  /** Returns whether this {@link CompletableResultCode} has completed. */
+  public boolean isDone() {
+    synchronized (lock) {
+      return succeeded != null;
+    }
+  }
+
+  /**
+   * Waits for the specified amount of time for this {@link CompletableResultCode} to complete. If
+   * it times out or is interrupted, the {@link CompletableResultCode} is failed.
+   *
+   * @return this {@link CompletableResultCode}
+   */
+  public CompletableResultCode join(long timeout, TimeUnit unit) {
+    synchronized (lock) {
+      if (succeeded != null) {
+        return this;
+      }
+    }
```

Comment:
same comment as on @anuraaga 's PR: can we just do `if (isDone()) return this;`  instead of duplicating the code?

Reply:
Rebased and took all changes from Anuraag PR